app/features/receive/receive-cashu-token-hooks.tsx

```diff
@@ -0,0 +1,312 @@
+import type { Token } from '@cashu/cashu-ts';
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import type {
+  Account,
+  CashuAccount,
+  ExtendedCashuAccount,
+} from '~/features/accounts/account';
+import {
+  useAccount,
+  useAccounts,
+  useAddCashuAccount,
+  useDefaultAccount,
+} from '~/features/accounts/account-hooks';
+import { tokenToMoney } from '~/features/shared/cashu';
+import { getClaimableProofs, getUnspentProofsFromToken } from '~/lib/cashu';
+import { checkIsTestMint, getMintInfo } from '~/lib/cashu';
+import { useLatest } from '~/lib/use-latest';
+import type { AccountWithBadges } from '../accounts/account-selector';
+import {
+  useCreateCashuTokenSwap,
+  useTokenSwap,
+} from './cashu-token-swap-hooks';
+
+type CashuAccountWithBadges = AccountWithBadges<CashuAccount>;
+
+type UseGetClaimableTokenProps = {
+  token: Token;
+  cashuPubKey: string;
+};
+
+type TokenQueryResult =
+  | {
+      /** The token with only claimable proofs. Will be null if the token cannot be claimed */
+      claimableToken: Token;
+      /** The reason why the token cannot be claimed. Will never be defined when claimableToken is not null. */
+      cannotClaimReason: never;
```

Comment:
I don't like how we have to cast undefined to never now. Also looking at [the never docs](https://www.typescriptlang.org/docs/handbook/basic-types.html#never) I might have been wrong that it is intended for cases like this. Thus I would change this back to null